app/controllers/admin/api/service_contracts_controller.rb

```diff
@@ -21,14 +29,43 @@ def destroy
     respond_with(service_subscription.unsubscribe(service_contract))
   end
 
+  # Service Subscription Update
+  # PUT /admin/api/accounts/{account_id}/service_contracts/{id}.xml
+  def update
+    service = service_contract.issuer
+    new_plan = service.service_plans.find(service_contract_plan_id)
+    service_contract.change_plan!(new_plan)
+    respond_with(service_contract)
+  end
+
   protected
 
   def account
     @account ||= current_account.buyers.find params.require(:account_id)
   end
 
+  def find_service
+    @service = service
+  end
+
   def service_contract
     @service_contract ||= account.bought_service_contracts.find params.require(:id)
   end
 
+  def service_contract_params
+    params.permit(service_contract: [:plan_id])
+          .fetch(:service_contract).merge(plan: service_plan)
+  end
+
+  def service
+    service ||= accessible_services.find(params[:service_id])
+  end
+
+  def service_plan(plan_id = service_contract_plan_id)
+    service_plan ||= service.service_plans.find_by(id: plan_id)
```

Comment:
`service_plan` is a local variable, which is never reused. I guess you meant `@service_plan ||=... ` ?